synapse/api/filtering.py

```diff
@@ -444,6 +444,20 @@ def lazy_load_members(self):
     def include_redundant_members(self):
         return self.filter_json.get("include_redundant_members", False)
 
+    def with_room_ids(self, room_ids):
+        """Returns a new filter with the given room IDs appended.
+
+        Args:
+            room_ids (list): A list of room_ids.
```

Comment:
it will work with any iterable, and please specify the types of the entries:
```suggestion
            room_ids (iterable[unicode]): The room_ids to add
```